lib/features/modeling/BpmnUpdater.js

```diff
@@ -405,6 +410,11 @@ BpmnUpdater.prototype.updateDiParent = function(di, parentDi) {
     return;
   }
 
+  // Cover the case where di.$parent === undefined and parentDi === null
+  if (!parentDi && !di.$parent) {
+    return;
```

Comment:
When does this case occur and do we have a test case for it?

Reply:
This happens when we replace a Task with an expanded sub-process. The newly created Start event has a `di.$parent = null`.

We don't test this explicitly, but tests like this check if we can replace Tasks with expanded sub-processes https://github.com/bpmn-io/bpmn-js/blob/194b963959a526e8e0805043debf43bc7fb9fdb2/test/spec/features/modeling/behavior/SubProcessStartEventBehaviorSpec.js#L29-L47

Reply:
Thanks for that explaination.

But then, why did it not blow up in the past? I'll investigate and report back.

Reply:
I believe [this](https://github.com/bpmn-io/bpmn-js/pull/1473/files#diff-ba4080acef0bcddd7ff56277569d664b45c9089cb8f4d949ae4548c2320c57ebR112) is the issue. We cannot blindly re-use the existing DI on replace but must continue to create a new one.

Reply:
Otherwise the following will happen:

* Task (to be replaced) gets removed
* DI (of task _and_ replaced element) is being unlinked
* Replaced element does not have a proper DI reference anymore

Reply:
Tests for proper DI updating after replace (`BpmnReplaceSpec`) are missing entirely and need to be added.